src/compiler/transformers/classFields.ts

```diff
@@ -266,15 +266,44 @@ namespace ts {
 
         /**
          * If we visit a private name, this means it is an undeclared private name.
-         * Replace it with an empty identifier to indicate a problem with the code.
+         * Replace it with an empty identifier to indicate a problem with the code,
+         * unless we are in a statement position - otherwise this will not trigger
+         * a SyntaxError.
          */
         function visitPrivateIdentifier(node: PrivateIdentifier) {
             if (!shouldTransformPrivateElementsOrClassStaticBlocks) {
                 return node;
             }
+            if (isStatement(node.parent)) {
+                return node;
+            }

```

Comment:
This was added because we are now visiting privateIdentifiers that were previously not going through here. Existing tests caught this change. So this is here so to match the expectations of the existing baselines.

Specifically:

```ts
#x
```

Right now this invalid privateId is preserved in the output so there is a SyntaxError. If we transform it to the empty identifier then there is nothing left and the SyntaxError goes away. But that was only happening because `visitPrivateIdentifier` wasn't being called for code like that.

https://www.typescriptlang.org/play?#code/MQDyA

Another option is to always emit the privateId as that should be a SyntaxError anyway. Though maybe that should also be a seperate PR.